pxtcompiler/emitter/service.ts

```diff
@@ -833,10 +804,29 @@ namespace ts.pxtc.service {
         }
     }
 
+    type Py2TsRes = {
+        // TODO: use the real types from converter.ts
+        diagnostics: pxtc.KsDiagnostic[],
+        success: boolean,
+        outfiles: { [key: string]: string },
+        syntaxInfo?: pxtc.SyntaxInfo
+    }
+    function updatePySyntaxInfo(opts: CompileOptions) {
+        // TODO: We call py2ts here directly to get the python syntax info.
+        // We likely already have the syntax info from a previous compile,
+        // we should do better about cache the latest results and serving those
+        // here. Also, we shouldn't mutate "opts" (the input).
+        let res = (pxt as any).py.py2ts(opts) as Py2TsRes
```

Comment:
Why the `any` cast here?

Reply:
I forget the justification, but Michal tried for a while to get references to the python code to work right with TSC and I believe I gave up. I think I tried at one point too. If you have ideas, I'm all ears.